### PR TITLE
Updated getDBConnection and queryODBC to support snowflake

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Updated getDBConnection and queryODBC to support snowflake.

It seems the Unit test fails with exp_ts_cluster which has nothing to the PR.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/system.R [hunk 1]
```diff
@@ -1029,6 +1029,61 @@ getDBConnection <- function(type, host = NULL, port = "", databaseName = "", use
                              bigint = "numeric")
       connection_pool[[key]] <- conn
     }
+  } else if (type == "snowflake") {
+    # If the platform is Linux, set the below predefined driver installed on Collaboration Server
+    # so that this data source can be scheduled.
```
Comment (hideaki) [inline R/system.R:1034]: Add comment on how this works - it overwrites the value for desktop when run on server.

File: R/system.R [hunk 2]
```diff
@@ -1113,7 +1168,7 @@ clearDBConnection <- function(type, host = NULL, port = NULL, databaseName, user
 }
 
 isConnecitonPoolEnabbled <- function(type){
```
Comment (hideaki) [inline R/system.R:1170]: Correct typo.

File: R/system.R (hunk 1 above)
Comment (hidekoji) [reply R/system.R:1034]: Added comment.

File: R/system.R (hunk 2 above)
Comment (hidekoji) [reply R/system.R]: fixed typo.

**Pull request merged**